src/Microsoft.Framework.Configuration/ConfigurationSection.cs

```diff
@@ -65,14 +87,14 @@ public string Value
 
                 foreach (var src in Sources)
                 {
-                    src.Set(Key, value);
+                    src.Set(Path, value);
                 }
             }
         }
 
         protected override string GetPrefix()
         {
-            return _key + Constants.KeyDelimiter;
+            return Path + Constants.KeyDelimiter;
```

Comment:
BTW, this would imply moving the implementation of Path to ConfigurationBase, even if it is only defined in IConfigurationSection that should be fine.